sdk/python/feast/infra/registry/sql.py

```diff
@@ -178,10 +179,19 @@ class FeastMetadataKeys(Enum):
 )
 
 
+class SqlRegistryConfig(RegistryConfig):
+    registry_type: StrictStr = "sql"
+    """ str: Provider name or a class name that implements Registry."""
+
+    path: StrictStr = ""
+    """ str: Path to metadata store.
+    If registry_type is 'sql', then this is a database URL as expected by SQLAlchemy """
+
+
 class SqlRegistry(BaseRegistry):
     def __init__(
         self,
-        registry_config: Optional[RegistryConfig],
+        registry_config: Optional[SqlRegistryConfig],
```

Comment:
hey @davidschuler-8451  I have another idea, let's make this one `Union[RegistryConfig, SqlRegistryConfig]`